lib/java_buildpack/container/tomee/tomee_instance.rb

```diff
@@ -34,6 +44,21 @@ def tomcat_7_compatible
         @version < TOMEE_7
       end
 
+      private
+
+      def web_inf_lib
+        if ear?
+          # copy additional libraries to tomee lib folder
+          tomcat_lib
```

Comment:
Found a few more details, will add on as I discover more stuff - all tests are based on this sample(https://github.com/bijukunjummen/eartest)

Regarding: _Web module packages the driver in WEB-INF/lib and places resources.xml in WEB-INF_

I am finding that if you specify a driver in resources.xml, doesn't matter where (WEB-INF/resources.xml, META-INF/resources.xml) then the only way the driver is resolved is through tomee/lib folder, not via ear/lib. Shared resources(common models, interfaces etc) are resolved okay though, it is only the resources that are not resolving

The download of the driver behavior works same as with wars, if a driver is packaged with the app then it is not downloaded, however given the behavior with resources.xml, I am not really sure why a user would package the jar anyway :-)

Regarding _EJB module and Web module use common driver placed in EAR/lib; EJB module places resources.xml in it's META-INF (not the EAR's META-INF) and Web module places resources.xml in WEB-INF_

Assuming the driver is available(say with hsql), then it appears to work fine with this combination

Regarding _EJB module uses driver placed in the EAR file and specified in the EJB's META-INF/MANIFEST.MF via Class-Path header; EJB module places resources.xml in it's META-INF (not the EAR's META-INF)_

don't know how to try this yet, will get back.